nixos/modules/services/games/minecraft-server.nix

```diff
@@ -36,6 +73,34 @@ in
         '';
       };
 
+      # use https://mcuuid.net/ to get minecraft uuid for username
+      whitelist = mkOption {
+        type = types.nullOr (types.attrsOf (types.strMatching "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"));
+        default = {};
+        description = "Attribute set of { name = uuid }";
+        example = {
+          "username1" = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx";
+          "username2" = "yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy";
+        };
+      };
+
+      serverProperties = mkOption {
+        type = types.attrsOf (types.either types.bool (types.either types.int types.str));
+        default = {};
+        description = ''
+          server.properties to use for minecraft server
+          these attributes take priority and are merged with
+          defaultServerProperties
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.minecraft-server;
```

Comment:
Ahh, still the same issue, but this time with the default value. You need `defaultText = "pkgs.minecraft-server"` as well. (most likely)

Reply:
Fixed.